### DISCUSSION TITLE
Plugins

### DISCUSSION THREAD

**Discussion by katieamazing** [testingisdocumenting/znai#1044] [General]
Hi Nick, I was curious about plugin development for Znai.

I am interested in creating a simple reader feedback form that might live in a header/footer of Znai pages and log responses somewhere. Is there any similar prior work? Can you tell me a little about the plugin development process to help me understand the scope of work and the pros/cons of internal vs. external?

Thank you!

**Comment by MykolaGolubyev** [comment]
Hi Katie! Long time. Thanks for reaching out.
It is a good topic you brought up.

I have been meaning to document how to write plugins but there were no demand so far.

Is your goal to have the Feedback controls on every page or you want to selectively control the pages?

Plugins so far are only content based. I.e. you explicitly use them to add new content to the place of plugin usage. 

If it helps, we can voice chat online or offline to streamline the discussion. Ping me on LinkedIn and we can set something up.

**Reply by katieamazing** [reply]
My vision (which is pretty flexible!) is to have a display widget on the top of every page next to the presentation mode icon. And to have the interactable feedback UI inside the footer on each page.

I'll find you on LinkedIn and we can connect more real-time :)

**Reply by MykolaGolubyev** [reply]
@katieamazing sounds useful! followed you on LinkedIn
